process/block/preprocess/transactionsV2.go

```diff
@@ -0,0 +1,491 @@
+package preprocess
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage/txcache"
+)
+
+// TxType identifies the type of the tx
+type TxType int32
+
+const (
+	nonScTx TxType = 0
+	scTx    TxType = 1
+)
+
+type processedTxsInfo struct {
+	numTxsAdded                        int
+	numTxsBad                          int
+	numTxsSkipped                      int
+	numTxsFailed                       int
+	numTxsWithInitialBalanceConsumed   int
+	numCrossShardScCallsOrSpecialTxs   int
+	totalTimeUsedForProcess            time.Duration
+	totalTimeUsedForComputeGasConsumed time.Duration
+}
+
+func (txs *transactions) createAndProcessMiniBlocksFromMeV2(
```

Comment:
this function is too long. can we split again ?

Reply:
Will try to split in the latest PR, as if it will be done in this one it will affect/conflict with all the others which started with this base